addons/mail/static/src/core/web/activity_menu.js

```diff
@@ -55,7 +59,12 @@ export class ActivityMenu extends Component {
             // So, duplicates are faking the count and "Load more" doesn't show up
             force_search_count: 1,
         };
-        let domain = [["activity_user_id", "=", this.userId]];
+        let domain;
+        if (group.model !== "mail.activity") {
+            domain = [["activity_user_id", "=", this.userId]];
+        } else {
+            domain = [["id", "in", group.activity_ids]];
+        }
         if (group.domain) {
```

Comment:
Sorry, I don’t see your point here. Actually for “other activities”, we compute in advance which activities to display while for others we use a domain ([["activity_user_id", "=", this.userId]]). Here with the new implementation using an server side defined action, as you have suggested, it should be simpler.

Reply:
What I meant here is to do in the python:

`user_activities['mail.activity']['domain'] = [('id', 'in', activities.ids)]`

And let this JS bit handle that "domain" building by itself, since it's apparently already supported.
(Adding the user_id part should not change anything).

You would have to move your new code above a bit lower if so, and use the domain instead of active_ids.
Not a huge deal anyway, the idea was to keep a bit the same structure as what we already have for other models.

Reply:
I will keep it like this for now then as I think it is not possible anymore with the new version using an action defined on the server side as we now use “active_ids” to transmit the group.activity_ids to the action through the context. I think we cannot force a domain with doAction but maybe I miss a feature.